### PR TITLE
check for empty string in use_crate()

### PR DESCRIPTION
**Pull request by kbvernon**
To address https://github.com/extendr/rextendr/issues/387, this PR adds a check for empty strings when running `purrr::discard()` on the `cargo_add_opts` list.

### CODE REVIEW

**Comment by CGMossa** [comment]
How do we know that this works? Is it possible to add a test? Otherwise I'd just merge.

**Comment by kbvernon** [comment]
yes, this function needs a test. i'll look into adding one tomorrow!

**Comment by kbvernon** [comment]
Pasting flags and values together into single character scalars is the wrong way to pass additional arguments to `processx::run()`, so I opted to concatenate them into a character vector. This has the advantage of making `purrr::discard()` unnecessary, as concatenating drops NULL values.

File: tests/testthat/test-use_crate.R [hunk 1]
```diff
@@ -0,0 +1,28 @@
+test_that("use_crate() adds dependency to package or workspace", {
```
Comment (Ilia-Kosenkov) [inline tests/testthat/test-use_crate.R:1]: Could you please add a case for unhappy path? E.g. both version and git?

Comment (kbvernon) [reply tests/testthat/test-use_crate.R:1]: i also added the "rextendr_error" class to the `cli_abort()` call

**Comment by JosiahParry** [comment]
@kbvernon one last test! can you test the optional feature? The nifty codecov tool shows that it is missing a check :)

**Comment by JosiahParry** [comment]
100% code cov? You're crazy for that.

**Review by JosiahParry — APPROVED** [review]

**Pull request merged**